src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java

```diff
@@ -125,17 +132,26 @@ public Leaderboard fromString(String string) {
     };
   }
 
-  public void onLeaderboardSelected() {
-    contentPane.setVisible(false);
-    leaderboardService.getEntries(leaderboardComboBox.getValue()).thenAccept(leaderboardEntryBeans -> {
-      ratingTable.setItems(observableList(leaderboardEntryBeans));
-      contentPane.setVisible(true);
-    }).exceptionally(throwable -> {
+  public ChangeListener<Leaderboard> onLeaderboardSelected() {
+    return (obs, oldValue, newValue) -> {
       contentPane.setVisible(false);
-      log.warn("Error while loading leaderboard entries", throwable);
-      notificationService.addImmediateErrorNotification(throwable, "leaderboard.failedToLoad");
-      return null;
-    });
+      searchTextField.clear();
+      if (usernamesAutoCompletion != null) {
+        usernamesAutoCompletion.dispose();
+      }
+      leaderboardService.getEntries(newValue).thenAccept(leaderboardEntryBeans -> {
+        ratingTable.setItems(observableList(leaderboardEntryBeans));
+        usernamesAutoCompletion = TextFields.bindAutoCompletion(searchTextField,
+            leaderboardEntryBeans.stream().map(LeaderboardEntry::getUsername).collect(Collectors.toList()));
+        usernamesAutoCompletion.setDelay(0);
```

Comment:
Don't you think this is a bad thing that the client asks the api with a request on the user table (big table) every letter you type in. Say the username is TrustTheFall. Makes 12 http requests and sql queries XD

Reply:
I mean their should definitely be a big delay to ensure the server does not get too many requests

Reply:
This action is on the comboBox choice not the textField input I believe

Reply:
So the api is only called when the user changes the leaderboard they are searching. Then that list is used to populate the autocompletion.

Reply:
Then this only work with first 10 000 again ... well ok

Reply:
Please call oldValue and newValue with proper names then I would probably better understand what you are doing. newValue-> newlySelectedLeaderBoard